lldb/include/lldb/Utility/AnsiTerminal.h

```diff
@@ -171,7 +171,45 @@ inline std::string FormatAnsiTerminalCodes(llvm::StringRef format,
   }
   return fmt;
 }
+
+inline std::string StripAnsiTerminalCodes(llvm::StringRef str) {
+  std::string stripped;
+  while (!str.empty()) {
+    llvm::StringRef left, right;
+
+    std::tie(left, right) = str.split(ANSI_ESC_START);
+    stripped += left;
+
+    // ANSI_ESC_START not found.
+    if (left == str && right.empty())
+      break;
+
+    auto end = llvm::StringRef::npos;
+    for (size_t i = 0; i < right.size(); i++) {
+      char c = right[i];
+      if (c == 'm' || c == 'G') {
+        end = i;
+        break;
+      }
+      if (isdigit(c) || c == ';')
+        continue;
+
+      break;
+    }
+
+    // ANSI_ESC_END not found.
+    if (end != llvm::StringRef::npos) {
+      str = right.substr(end + 1);
+      continue;
+    }
+
+    stripped += ANSI_ESC_START;
+    str = right;
```

Comment:
I think it should be equivalent, but a little less lumberjacky.

```suggestion
    size_t end = right.find_first_not_of("0123456789;");
    if (end < right.size() && (right[end] == 'm' || right[end] == 'G')) {
      str = right.substr(end + 1);
    } else {
      // ANSI_ESC_END not found.
      stripped += ANSI_ESC_START;
      str = right;
    }
```

Reply:
My code being called "lumberjacky" is probably the most beautiful insult I've ever received 😂

Reply:
Because your code sleeps all night and works all day?